### PR TITLE
Add seperate docs and landing layouts.

### PR DESCRIPTION
**Pull request by notlee**
- Add modifier class `o-layout--docs` for the documentation layout (currently default).
- Add modifier class `o-layout--landing` to house the upcoming landing page layout (e.g. homepage).
- Use `o-typography` spacing size for consistent verticle rhythm.
- Remove "deprecated" class `o-layout__main--full-span`.
- Update documentation layout demo.

The documentation layout demo has been updated. It's smaller and more focused on features of the layout itself. The aside uses a semantic heading level which looks visually wrong, but will look better when [o-typography@^5.8.0](https://github.com/Financial-Times/o-typography/pull/166) is released.
screenshot 2018-12-18 at 15 35 31

The landing layout is just starting out (a hero and overview area will be added, and content within main will have an appropriate line length).
screenshot 2018-12-18 at 15 35 25

### CODE REVIEW

File: src/scss/_grid.scss [hunk 1]
```diff
@@ -8,8 +8,98 @@
...
+			"main"
+			"footer";
+	}
+}
+
+
+/// Grid for a landing page layout (e.g. homepage).
+/// @access private
+@mixin _oLayoutLandingGrid() {
+	.o-layout.o-layout--landing {
```
Comment (gvonkoss) [inline src/scss/_grid.scss]: do we need this specificity?

File: src/scss/_grid.scss [hunk 2]
```diff
@@ -8,8 +8,98 @@
...
+		}
+
+		> * {
+			grid-column: 1;
+		}
+
+		@include oGridRespondTo($from: L) {
+			> aside {
+				grid-column: 2;
+				grid-row: span 3; //hack mchack, need to look into spanning across rows irrespective of siblings
```
Comment (gvonkoss) [inline src/scss/_grid.scss]: 😂 i'm amused this has remained

Comment (notlee) [reply src/scss/_grid.scss]: 😂 I'll see if there's anything we can do to remove hack mchack.

File: src/scss/_grid.scss (hunk 1 above)
Comment (notlee) [reply src/scss/_grid.scss]: We can probably rely on source order but there's no reason not to be explicit (the specificity is still low).

**Pull request merged**